main.c

```diff
@@ -4631,9 +4631,12 @@ ovl_rename_direct (fuse_req_t req, fuse_ino_t parent, const char *name,
       ret = direct_renameat2 (srcfd, name, destfd,
                               newname, flags|RENAME_WHITEOUT);
     }
-      /* If the destination is a whiteout, just overwrite it.  */
-  if (ret < 0 && errno == EEXIST)
-    ret = direct_renameat2 (srcfd, name, destfd, newname, flags & ~RENAME_NOREPLACE);
+  /* If the destination is a whiteout, just overwrite it.  */
+  if (ret < 0 && errno == EEXIST) {
+    ret = direct_renameat2(srcfd, name, destfd, newname, flags & ~RENAME_NOREPLACE);
+    if (!(ret < 0))
```

Comment:
this looks weird.

Could we have `ret >= 0`?

Reply:
fixed.